edb/pgsql/delta.py

```diff
@@ -2251,17 +2358,19 @@ def apply(self, schema, context):
                 self.schedule_post_inhview_update_command(
                     schema,
                     context,
-                    (lambda nschema, ncontext:
-                     self.enforce_constraint(
-                         constraint, nschema, self.span
-                     )
-                     if nschema.has_object(constraint.id)
-                     else None),
+                    (
+                        lambda nschema, ncontext:
+                            self.enforce_constraint(
+                                constraint, nschema, self.span
+                            )
+                            if nschema.has_object(constraint.id)
+                            else None
+                    ),
```

Comment:
I think that we might want to schedule these `enforce_constraint`s to happen at the end, at the same time we create the triggers.

Since migrations can contain DML queries, we want to avoid the situation where a conflicting object is inserted *before* we create the trigger. I think that doing the enforce check when we create the trigger might solve that problem?

Reply:
Oh, actually nevermind about that, since the trigger updates get run at the same time as post_inhview updates.

Reply:
Ah also, this was me reformatting that code since during one of my attempts I ended up doing some pretty unreasonable things with the post inhview commands.